scripts/add.tokens.erc20.mjs

```diff
@@ -0,0 +1,377 @@
+	if (match) {
+		const list = match[1].trim();
+		if (!list.includes(element)) {
+			const updatedList = list ? `${list}, ${element}` : element;
+			return {
+				content: content.replace(regex, match[0].replace(match[1], updatedList)),
+				hasChanged: true
+			};
+		}
+		return { content, hasChanged: false };
+	}
+	const newLine = unescapeRegExp(regex.source.replace('([\\s\\S]*?)', element));
+	return {
+		content: addAtStartIfNotFound ? `${newLine}\n${content}` : `${content}\n${newLine}`,
+		hasChanged: true
+	};
+};
+
+const updateListInContent = ({ content, regex, elements, addAtStartIfNotFound = false }) =>
+	elements.reduce(
+		(acc, element) => {
+			const { content: newContent, hasChanged } = addElementListInContent({
+				content: acc.content,
+				regex,
+				element,
+				addAtStartIfNotFound
+			});
+			return {
+				content: newContent,
+				hasChanged: acc.hasChanged || hasChanged
+			};
+		},
+		{ content, hasChanged: false }
+	);
+
+const updateImportsInContent = ({ content, imports, module }) =>
+	updateListInContent({
+		content,
+		regex: new RegExp(`import \\{([\\s\\S]*?)} from '${escapeRegExp(module)}';`),
+		elements: imports,
+		addAtStartIfNotFound: true
+	});
+
+const fetchTokenDetails = async ({ contractAddress, isTestnet }) => {
+	const provider = new ethers.providers.EtherscanProvider(
+		isTestnet ? 'sepolia' : 'homestead',
+		ETHERSCAN_API_KEY
+	);
+	const contract = new ethers.Contract(
+		contractAddress,
+		['function name() view returns (string)', 'function decimals() view returns (uint8)'],
+		provider
+	);
+
+	const [name, decimals] = await Promise.all([contract.name(), contract.decimals()]).catch(
+		(err) => {
+			console.error(`Error fetching token details:\n${err}`);
+			process.exit(1);
+		}
+	);
+
+	return { name, decimals };
+};
+
+const loadFileContentOrEmpty = (filePath) => {
+	try {
+		return readFileSync(filePath, 'utf8');
+	} catch (err) {
+		console.log(`File ${filePath} does not exist, it will be created.`);
+		return '';
+	}
+};
+
+let filesCreatedOrModified = false;
+
+const manageEnvFile = async ({ mainSymbol: symbol, contractAddress, testnetContractAddress }) => {
+	const fileName = `tokens.${symbol.toLowerCase()}.env.ts`;
+	const filePath = path.join(DATA_DIR_PATH, fileName);
+
+	const existingFileContent = loadFileContentOrEmpty(filePath);
+
+	const mainnetToken = nonNullish(contractAddress) ? `${symbol}_TOKEN` : undefined;
+	const testnetToken = nonNullish(testnetContractAddress) ? `SEPOLIA_${symbol}_TOKEN` : undefined;
+
+	const mainnetTokenCreated = existingFileContent.split(/\W+/).includes(mainnetToken ?? '');
+	const testnetTokenCreated = existingFileContent.split(/\W+/).includes(testnetToken ?? '');
+
+	if (mainnetTokenCreated && testnetTokenCreated) {
+		console.log(`Variables for token ${symbol} already exists in the environment file ${fileName}`);
+		return { fileName, mainnetToken: undefined, testnetToken: undefined };
+	}
+
+	const { name, decimals } = await fetchTokenDetails({
```

Comment:
Do I understand correctly that you are fetching the token details either for production or testnets? If correct, isn't this inaccurate? Can't token information, notably name, be potentially different on mainnet and testnet? Shouldn't we always fetch their corresponding information?

Reply:
indeed, I was incorrectly assuming they would be the same, so I wanted to reduce the fetching time.

But it is not the same (as you pointed out for example, USD has different names for Mainnet and Sepolia), so I will change it